packages/cmd/src/syntax/SyntaxValidator.ts

```diff
@@ -294,8 +294,24 @@ export class SyntaxValidator {
 
                 // check if the value of the option conforms to the allowableValues (if any)
                 if (!isNullOrUndefined(optionDef.allowableValues)) {
-                    if (!this.checkIfAllowable(optionDef.allowableValues, commandArguments[optionName])) {
-                        this.invalidOptionError(optionDef, responseObject, commandArguments[optionName]);
+                    // Make a copy of optionDef, so that modifications below are only used in this place
+                    const optionDefCopy: ICommandOptionDefinition = JSON.parse(JSON.stringify(optionDef));
+                    // Use modified regular expressions for allowable values to check and to generate error
+                    optionDefCopy.allowableValues.values = optionDef.allowableValues.values.map((regex) => {
+                        // Prepend "^" if not existing
+                        if (!regex.startsWith("^")) {
+                            regex = "^" + regex;
+                        }
+
+                        // Append "$" if the last char is an escaped "$" or chars other than "$"
+                        if (!regex.endsWith("$") || regex.endsWith("\\$")) {
```

Comment:
Could we maybe add a test around this (or copy an existing one and modify the data)? i.e. pass the following string: `^abc\\$$` 😄

Reply:
And maybe don't rely on snapshots for a regular string. (`abc`) :smile:

Reply:
Now I have added more tests on allowable values. Have a look and please let me know if there's something I can help.

Reply:
Awesome! 
Clever way of adding more permutations.

**Before:**
![image](https://user-images.githubusercontent.com/37381190/78561556-efa4ea00-77e5-11ea-9c06-8dbfc5006c97.png)

**After:**
![image](https://user-images.githubusercontent.com/37381190/78561605-06e3d780-77e6-11ea-9695-b1316310c82b.png)